test/functional/ui/cmdline_spec.lua

```diff
@@ -857,4 +859,76 @@ describe("cmdline height", function()
     screen:try_resize(25, 5)
     assert_alive()
   end)
+
+  it('cmdheight can be 0', function()
+    clear()
+    local screen = Screen.new(25, 5)
+    screen:attach()
+
+    command("set cmdheight=1 noruler laststatus=2")
+    screen:expect{grid=[[
+      ^                         |
+      ~                        |
+      ~                        |
+      [No Name]                |
+                               |
+    ]]}
+
+    command("set cmdheight=0 noruler laststatus=2")
+    screen:expect{grid=[[
+      ^                         |
+      ~                        |
+      ~                        |
+      ~                        |
+      [No Name]                |
+    ]]}
+
+    command("set cmdheight=0 ruler laststatus=0")
+    screen:expect{grid=[[
+      ^                         |
+      ~                        |
+      ~                        |
+      ~                        |
+      ~                        |
+    ]]}
+
+    command("set cmdheight=0 noruler laststatus=0 showmode")
+    feed('i')
+    screen:expect{grid=[[
+      ^                         |
+      ~                        |
+      ~                        |
+      ~                        |
+      ~                        |
+    ]], showmode={}}
+    feed('<Esc>')
+    eq(0, eval('&cmdheight'))
+
+    command("set cmdheight=1 noruler laststatus=0 showmode")
+    feed('i')
+    screen:expect{grid=[[
+      ^                         |
+      ~                        |
+      ~                        |
+      ~                        |
+      -- INSERT --             |
+    ]]}
+    feed('<Esc>')
+    eq(1, eval('&cmdheight'))
+
+    command("set cmdheight=0 noruler laststatus=0")
+    feed(':')
+    screen:expect{grid=[[
+                               |
+      ~                        |
+      ~                        |
+      ~                        |
+      :^                        |
```

Comment:
test should also feed some input into the cmdline. That's the most important part of this feature...

Reply:
Thanks.  I have added the tests.

Reply:
Did you add "feed some input"?